doc.go

```diff
@@ -9,6 +9,8 @@ var ErrNotFound = errors.New("not found")
 // Geche interface is a common interface for all cache implementations.
 type Geche[K comparable, V any] interface {
 	Set(K, V)
+	// SetIfPresent sets the kv only if the key was already present, and returns whether the insertion was performed
+	SetIfPresent(K, V) bool
```

Comment:
Oh, I just realised that it would be a shame to implement atomic Get/Set without returning the previous value. This will increase the number of possible use cases.

Reply:
Good idea, I'll do this later

Reply:
refactored